tensorflow/core/graph/mkl_layout_pass.cc

```diff
@@ -2810,6 +2837,73 @@ class MklLayoutRewritePass : public GraphOptimizationPass {
     return n;
   }
 
+  // Find Relu or Conv2D node that can be merged with input node 'm'.
+  // If input 'm' is Relu, then check if there exists Conv2D node that can be
+  // merged with 'm'. If input 'm' is Conv2D, then check if there exists Relu
+  // node that can be merged with 'm'.
+  static Node* GetConv2DOrRelu(const Node* m) {
+    CHECK_NOTNULL(m);
+    Node* n = nullptr;
+
+    if (m->type_string() == csinfo_.relu) {
+      // If a is Relu, then Conv2D is 0th input of Relu.
+      TF_CHECK_OK(m->input_node(0, &n));
+    } else {
+      CHECK_EQ(m->type_string(), csinfo_.conv2d);
+      // Go over all output edges and search for Relu Node.
+      // 0th input of Relu is Conv2D.
+      for (const Edge* e : m->out_edges()) {
+        if (!e->IsControlEdge() && e->dst()->type_string() == csinfo_.relu &&
+            e->dst_input() == 0) {
+          n = e->dst();
+          break;
+        }
+      }
+    }
+
+    if (n == nullptr) {
+      VLOG(1) << "MklLayoutRewritePass: Could not find matching "
+              << "Conv2D and Relu node for merging. Input node: "
+              << m->DebugString();
+    }
+
+    return n;
+  }
+
+  // Find Conv2DWithBias or Relu node that can be merged with input node 'm'.
+  // If input 'm' is Conv2DWithBias, then check if there exists Relu node that
+  // can be merged with 'm'. If input 'm' is Relu, then check if there exists
+  // Conv2DWithBias node that can be merged with 'm'.
+  static Node* GetConv2DWithBiasOrRelu(const Node* m) {
```

Comment:
This method is identical to GetConv2DOrRelu except that it uses csinfo_.conv2d_with_bias instead of csinfo_.conv2d and the error message. Please use a single method instead that passes the name of the convolution opeation as an argument.